### ISSUE TITLE
Suppress translation from validate:all command 

### ISSUE THREAD

**Issue by wannevancamp** [boxblinkracer/phpunuhi#42]
The `php vendor/bin/phpunuhi validate:all` command validates all translations. Is there any possibility of ignoring or suppressing some snippets from this check?

Several translations have been intentionally left blank. For example `shopBE.general.socials.pinterestLink` has no translation for en-GB but has a translation for it-IT.

![image](https://github.com/boxblinkracer/phpunuhi/assets/3399877/960a2459-082f-43b6-bf43-8d54fc999f81)

**Comment by boxblinkracer** [comment]
Hi

validate:all is more about validating snippets, code coverage, rules and so on

you can specifiy a specific translation set to be validated (--set)

but yes all translations across all locales will be validated, because thats the main thing with this framework
to figure out what is missing

what you could do is to use coverages..then you can say en-GB needs 100% while it-IT only needs 50%
but remember that (technically) softens the validation checks.....

**Comment by boxblinkracer** [comment]
im now more wondering why they are intentionally left blank?

**Comment by boxblinkracer** [comment]
one more question
could it be that you just need the "validate structure" part in your case

then i could add a "validate:structure" command for you

that would be the special use case where you dont care about the content, but only about everything existing or not?

**Comment by wannevancamp** [comment]
This is of course a rare case.

For an international shop, the social links can differ from language. In this case, there is a Pinterest account (and link) for some languages and not for others. That's why the snippet is left black for these languages. 

A "validate structure" command can help but then we are not sure that the snippets which need a translation are filled in.

**Comment by boxblinkracer** [comment]
aaaaaa

ok so you want to check everything but tell the system to ignore content of a few snippet keys?
only in a locale or in general?

i could create a new rule then you can specify something. 
so you can say ignore the contents of these keys. or contents of these keys in these locales. or a rule allow being empty. or allow being empty in these locales. 

what do you want to have?

**Comment by wannevancamp** [comment]
In our case we only need a rule `allow being empty in these locales of these keys` but I can imagine that the other rules can be helpfull for others.

**Comment by boxblinkracer** [comment]
ok perfect
so yeah i need a bit time to have a good concept
but basically you need something like

```xml 
<rules>
    <allowEmpty> 
           <key name="social.youtube.link">
                    <locale>EN</locale>
                   <locale>IT</locale>
           </key>
   </allowEmpty>
</rules>
```

yeah i know this is not the most efficient one
but just to double check my understanding of it for now :)

**Comment by wannevancamp** [comment]
Looks good. That's what we are looking for.
It is just an idea to use the TypeIdentifiers from the validators as an XML element?

```
<rules>
    <EMPTY_CONTENT>
        <key name="social.facebook.link"/>
        <key name="social.youtube.link">
            <locale>EN</locale>
            <locale>IT</locale>
        </key>
    </EMPTY_CONTENT>
    <CASE_STYLE>
        <key name="email.contact.subject"/>
    </CASE_STYLE>
</rules>
```

**Comment by boxblinkracer** [comment]
yeah not so much fan of upper case tags...it just looks bad somehow haha
and most of the time i try to not mix internal key identifiers and external user interfaces.
but i like they basic idea on that xml snippet.....ill check it out, but gimme a bit time please, pretty busy week...

**Comment by boxblinkracer** [comment]
hi there

so i've added the new empty content rule in the latest dev-master
can you see if it works for you?

everything is in the README :)

thank you

**Comment by wannevancamp** [comment]
Wow, that's very cool!
Thanks @boxblinkracer, for implementing this so fast. 👏

I've tested the different rules with different (wildcard) locales, it works perfectly.

**Comment by boxblinkracer** [comment]
perfect

thanks a lot for the great idea
so i double check on mjml and then release time 🤗

**Comment by boxblinkracer** [comment]
released in 1.17.0
